client/gutenberg/extensions/markdown/jetpack-markdown-block-editor.jsx

```diff
@@ -20,35 +18,32 @@ import MarkdownRenderer from './components/markdown-renderer';
 const PANEL_EDITOR = 'editor';
 const PANEL_PREVIEW = 'preview';
 
+/* eslint-disable wpcalypso/jsx-classname-namespace */
 class JetpackMarkdownBlockEditor extends Component {
-	constructor() {
-		super( ...arguments );
-
-		this.updateSource = this.updateSource.bind( this );
-		this.showEditor = this.showEditor.bind( this );
-		this.showPreview = this.showPreview.bind( this );
-		this.isEmpty = this.isEmpty.bind( this );
-
-		this.state = {
-			activePanel: PANEL_EDITOR,
-		};
-	}
+	state = {
+		activePanel: PANEL_EDITOR,
+	};
 
 	isEmpty() {
 		const source = this.props.attributes.source;
 		return ! source || source.trim() === '';
 	}
 
-	updateSource( source ) {
-		this.props.setAttributes( { source } );
-	}
+	updateSource = source => this.props.setAttributes( { source } );
 
-	showEditor() {
-		this.setState( { activePanel: 'editor' } );
-	}
+	toggleMode = mode => () => this.setState( { activePanel: mode } );
 
-	showPreview() {
-		this.setState( { activePanel: 'preview' } );
+	renderToolbarButton( mode, label ) {
+		const { activePanel } = this.state;
+
+		return (
+			<button
+				className={ `components-tab-button ${ activePanel === mode ? 'is-active' : '' }` }
+				onClick={ this.toggleMode( mode ) }
+			>
+				<span>{ label }</span>
```

Comment:
Is this span needed?

Reply:
Good question. Yes, the particular tab buttons use the spans for styling (same for the Custom HTML block).

Reply:
Got it, I had forgotten.